table.cc

```diff
@@ -2053,7 +2052,8 @@ table::mutation_query(schema_ptr s,
     std::exception_ptr ex;
   try {
     auto rrb = reconcilable_result_builder(*s, cmd.slice, std::move(accounter));
-    auto r = co_await q.consume_page(std::move(rrb), cmd.get_row_limit(), cmd.partition_limit, cmd.timestamp, class_config.max_memory_for_unlimited_query);
+    assert(cmd.max_result_size); // TODO: how do we know it's engaged?
```

Comment:
Was added to IDL in 92a7b16cba437501a02394e19c58f34f50ffa0be (4.3). So upgrading from 4.2 to 4.3 could break this.

Please split off into a separate minimal fix for backporting.

Reply:
(/cc @denesb)

Reply:
It's already a separate commit so should be easy to backport, but maybe I should replace the assertion with using a default value if this can break upgrades (also this probably means UB in master right now - since we use *cmd.max_result_size? unless it was added post-4.3 and we assume that we never upgrade from 4.2 to 4.4/higher)

Reply:
It needs an issue and it needs to be treated separately from this long series, it's too confusing.

Reply:
We know because all entry paths in the replica-side storage proxy set it if it came unset from the coordinator (coordinator is an old node).

Reply:
See https://github.com/scylladb/scylla/blob/dfc135dbd1944b0ed4a98980780139770514588e/service/storage_proxy.cc#L5020-L5023

Reply:
Also https://github.com/scylladb/scylla/blob/dfc135dbd1944b0ed4a98980780139770514588e/service/storage_proxy.cc#L5051-L5053